normandy/classifier/tests/test_api.py

```diff
@@ -52,3 +55,22 @@ def test_it_filters_by_locale(self, client):
 
         recipe_names = set(r['name'] for r in res.data['recipes'])
         assert recipe_names == {'german', 'any', 'both'}
+
+    @pytest.mark.xfail
+    def test_it_makes_no_db_queries(self, client, django_cache):
+        # Prepare some interesting data
+        recipe_action = RecipeActionFactory(recipe__enabled=True)
+        recipe = recipe_action.recipe
+
+        # Warm up the cache
+        res1 = client.post('/api/v1/fetch_bundle/', {'locale': 'de'})
+        assert res1.status_code == 200
+        assert res1.data['recipes'][0]['name'] == recipe.name
+
+        # Fire!
+        queries = CaptureQueriesContext(connection)
+        with queries:
+            res2 = client.post('/api/v1/fetch_bundle/', {'locale': 'de'})
+            assert res2.status_code == 200
+        assert res1.data == res2.data
+        assert (len(queries), list(queries)) == (0, [])
```

Comment:
Oh, this was mostly for debugging. Doing it this way showed me both a counter of how many tests were remaining, and also what the problematic queries were. I think this should just be comparing to the empty list (which gives really useful debugging when the test fails).